mitodl/micromasters
# edX user cached objects refactoring

#### What are the relevant tickets?
closes #1664
closes #1667 

#### What's this PR do?
Refactors the edX cached elements. Now only elements (enrollments, certificates, current grades) that are available for the user are cached. There are no more cached elements with `data=None`.
The calls for cache updates are still done synchronously during a request to the dashboard REST API

#### Where should the reviewer start?
`dashboard.api_edx_cache`

#### How should this be manually tested?
Run the migrations and verify that the dashboard rest api gives the same results as before.
Make sure that a payment for a course still results in a new cached enrollment. 
Make sure that the refresh via celery beat still works.

#### Any background context you want to provide?
This PR does not implement the asynchronous refresh of the edx user data.

NOTE: the tests in `api_edx_cache_test` are conceptually different from the tests removed from `api_test`, so do not try to match them.
